probatus/feature_elimination/feature_elimination.py

```diff
@@ -354,19 +369,60 @@ def fit(self, X, y, column_names=None):
         if self.random_state is not None:
             np.random.seed(self.random_state)
 
+        # If to columns_to_keep is not provided, then initialise it by an empty string.
+        # If provided check if all the elements in columns_to_keep are of type string.
+        if columns_to_keep is None:
+            len_columns_to_keep = 0
+        else :
+            if all(isinstance(x,str) for x in columns_to_keep):
+                len_columns_to_keep = len(columns_to_keep)
+            else :
+                raise(ValueError('The current values of columns_to_keep are not allowed.All the elements should be strings.'))
+        
+        # If the column_names parameters is provided, check if the elements in both columns_to_keep & column_names are matching.
+        if column_names is not None and columns_to_keep is not None :
+            if all(x in column_names for x in columns_to_keep):
+                pass
+            else :
+                raise(ValueError('The column names in parameter columns_to_keep and column_names are not macthing.'))
+
+        #Check that the total number of columns to select is less than total number of columns in the data.
+        #only when both parameters are provided.
+        if column_names is not None and columns_to_keep is not None :
+            if (self.min_features_to_select+len_columns_to_keep) > len(self.column_names):
```

Comment:
Are you sure we should do a plus here?

I would say a max between the two is enough:
Case 1: min=2, to_keep=1, overall we need at least 2 columns
Case 2: min =1, to_keep=2, overall we need at least 2
Case 3 min=1 and to_keep =1, overall we need 1
Case 4, min =0 to keep =5, overall we need 5
Case 5, min=5 to keep = 0, we need at least 5.

Also here don't check the column_names but the shape of X. I covered the explanation why in the previous comment

Reply:
Thanks, a max between the two can also work. I checked the about inputs as well. Works as expected.

Reply:
I agree